tests/components/lektrico/conftest.py

```diff
@@ -0,0 +1,148 @@
+
+MOCKED_DEVICE_ZC_NAME = "Lektrico-1p7k-500006._http._tcp"
+MOCKED_DEVICE_ZC_TYPE = "_http._tcp.local."
+MOCKED_DEVICE_ZEROCONF_DATA = ZeroconfServiceInfo(
+    ip_address=ip_address(MOCKED_DEVICE_IP_ADDRESS),
+    ip_addresses=[ip_address(MOCKED_DEVICE_IP_ADDRESS)],
+    hostname=f"{MOCKED_DEVICE_ZC_NAME.lower()}.local.",
+    port=80,
+    type=MOCKED_DEVICE_ZC_TYPE,
+    name=MOCKED_DEVICE_ZC_NAME,
+    properties={
+        "id": "1p7k_500006",
+        "fw_id": "20230109-124642/v1.22-36-g56a3edd-develop-dirty",
+    },
+)
+
+MOCKED_DEVICE_ZEROCONF_DATA_FOR_EM = ZeroconfServiceInfo(
+    ip_address=ip_address(MOCKED_DEVICE_IP_ADDRESS),
+    ip_addresses=[ip_address(MOCKED_DEVICE_IP_ADDRESS)],
+    hostname=f"{MOCKED_DEVICE_ZC_NAME.lower()}.local.",
+    port=80,
+    type=MOCKED_DEVICE_ZC_TYPE,
+    name=MOCKED_DEVICE_ZC_NAME,
+    properties={
+        "id": "m2w_810006",
+        "fw_id": "20230109-124642/v1.22-36-g56a3edd-develop-dirty",
+    },
+)
+
+MOCKED_DEVICE_ZEROCONF_DATA_FOR_3EM = ZeroconfServiceInfo(
+    ip_address=ip_address(MOCKED_DEVICE_IP_ADDRESS),
+    ip_addresses=[ip_address(MOCKED_DEVICE_IP_ADDRESS)],
+    hostname=f"{MOCKED_DEVICE_ZC_NAME.lower()}.local.",
+    port=80,
+    type=MOCKED_DEVICE_ZC_TYPE,
+    name=MOCKED_DEVICE_ZC_NAME,
+    properties={
+        "id": "m2w_830006",
+        "fw_id": "20230109-124642/v1.22-36-g56a3edd-develop-dirty",
+    },
+)
+
+MOCKED_DEVICE_BAD_ID_ZEROCONF_DATA = ZeroconfServiceInfo(
+    ip_address=ip_address(MOCKED_DEVICE_IP_ADDRESS),
+    ip_addresses=[ip_address(MOCKED_DEVICE_IP_ADDRESS)],
+    hostname=f"{MOCKED_DEVICE_ZC_NAME.lower()}.local.",
+    port=80,
+    type=MOCKED_DEVICE_ZC_TYPE,
+    name=MOCKED_DEVICE_ZC_NAME,
+    properties={
+        "id": "500006",
+        "fw_id": "20230109-124642/v1.22-36-g56a3edd-develop-dirty",
+    },
+)
+
+MOCKED_DEVICE_BAD_NO_ID_ZEROCONF_DATA = ZeroconfServiceInfo(
+    ip_address=ip_address(MOCKED_DEVICE_IP_ADDRESS),
+    ip_addresses=[ip_address(MOCKED_DEVICE_IP_ADDRESS)],
+    hostname=f"{MOCKED_DEVICE_ZC_NAME.lower()}.local.",
+    port=80,
+    type=MOCKED_DEVICE_ZC_TYPE,
+    name=MOCKED_DEVICE_ZC_NAME,
+    properties={"fw_id": "20230109-124642/v1.22-36-g56a3edd-develop-dirty"},
+)
+
+
+def _mocked_device_config() -> Settings:
+    return Settings(
+        type=MOCKED_DEVICE_TYPE,
+        serial_number=MOCKED_DEVICE_SERIAL_NUMBER,
+        board_revision=MOCKED_DEVICE_BOARD_REV,
+    )
+
+
+def _mocked_device_config_for_em() -> Settings:
+    return Settings(
+        type=MOCKED_DEVICE_TYPE_FOR_EM,
+        serial_number=MOCKED_DEVICE_SERIAL_NUMBER_FOR_EM,
+        board_revision=MOCKED_DEVICE_BOARD_REV,
+    )
+
+
+def _mocked_device_config_for_3em() -> Settings:
+    return Settings(
+        type=MOCKED_DEVICE_TYPE_FOR_3EM,
+        serial_number=MOCKED_DEVICE_SERIAL_NUMBER_FOR_3EM,
+        board_revision=MOCKED_DEVICE_BOARD_REV,
+    )
+
+
+def patch_device_config(device=None, exception=None):
```

Comment:
If you don't want to make this a fixture, remove the `device`, as it is not used, and make a new test helper which always raises, something like:
```py
def patch_device_config_with_error(exception):
    return patch(
        "homeassistant.components.lektrico.config_flow.Device.device_config",
        side_effect=exception,
    )
```

Reply:
I added fixtures, thanks.